Swiften/Elements/MIXJoin.h

```diff
@@ -43,16 +43,20 @@ namespace Swift {
                 jid_ = jid;
             }
 
-            const std::vector<MIXSubscribe::ref>& getSubscriptions() const {
+            const std::unordered_set<std::string>& getSubscriptions() const {
                 return subscribeItems_;
             }
 
-            void setSubscriptions(const std::vector<MIXSubscribe::ref>& value) {
+            void setSubscriptions(const std::unordered_set<std::string>& value) {
```

Comment:
This should be just `std::unordered_set<std::string>`. General rule, if you copy the parameter anyway, don't use `const T&`. For details see https://blogs.msdn.microsoft.com/xiangfan/2012/02/03/c98-c11-pass-by-value-or-pass-by-reference/ .

Also, the argument should be plural, so `values`.